### PR TITLE
Dom parser

### PR DESCRIPTION
**Pull request by asiajanicka**
XML with 5 classes (student group, student, grade, portal account, subject) + XSD
XmlUtils  - class with methods for parse/validate, etc
DOMTests - simple tests

### CODE REVIEW

File: src/main/java/org/example/utils/XmlUtils.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            schema = factory.newSchema(new File(pathToXSD));
+        } catch (SAXException e) {
+            e.printStackTrace();
+        }
+        return schema;
+    }
+
+    public static Document parseXml(String pathToXML) throws ParserConfigurationException, IOException, SAXException {
+        DocumentBuilderFactory factory
+                = DocumentBuilderFactory.newInstance();
```
Comment (vboltrukanis) [inline src/main/java/org/example/utils/XmlUtils.java]: Why do you move this to a second line, while line 43 is much longer.

File: src/test/java/org/example/parserTest/DOMTests.java [hunk 2]
```diff
@@ -41,6 +46,7 @@ public void validateXMLAgainstSchema() throws ParserConfigurationException, IOEx
                 .count())
                 .isEqualTo(2);
         assertThat(studentGroupsFromXML.get(1).getStudents().get(2).getGrades()).isEmpty();
+        logger.info("End of DOM Parser Tests - test case 1");
```
Comment (vboltrukanis) [inline src/test/java/org/example/parserTest/DOMTests.java]: Significant logging)

File: src/main/java/org/example/dao/GroupsHasTimetableEntriesDAO.java [hunk 3]
```diff
@@ -21,6 +23,12 @@ public class GroupsHasTimetableEntriesDAO implements IGroupsHasTimetableEntriesD
     private static final String CREATE_ENTRY = "INSERT INTO groups_has_time_table_entries (student_group_id," +
             " time_table_entry_id) VALUES (?, ?)";
     private static final String REMOVE_ENTRY = "DELETE FROM groups_has_time_table_entries WHERE id = ?";
+    private static final String REMOVE_ENTITY_FROM_GROUP = "DELETE FROM groups_has_time_table_entries WHERE student_group_id = ? " +
```
Comment (vboltrukanis) [inline src/main/java/org/example/dao/GroupsHasTimetableEntriesDAO.java:26]: Make sure you don't push unnecessary commits to a wrong branch

File: src/test/java/org/example/parserTest/DOMTests.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
+package org.example.parserTest;
+
+import org.example.model.StudentGroup;
+import org.example.utils.XmlUtils;
+import org.junit.jupiter.api.Test;
```
Comment (vboltrukanis) [inline src/test/java/org/example/parserTest/DOMTests.java:7]: Not the most popular lib, do you have a reason to choose exactly this one?

**Review by vboltrukanis — APPROVED** [review]

File: src/main/java/org/example/utils/XmlUtils.java (hunk 1 above)
Comment (asiajanicka) [reply src/main/java/org/example/utils/XmlUtils.java]: fixed

File: src/main/java/org/example/dao/GroupsHasTimetableEntriesDAO.java (hunk 3 above)
Comment (asiajanicka) [reply src/main/java/org/example/dao/GroupsHasTimetableEntriesDAO.java:26]: ok

File: src/test/java/org/example/parserTest/DOMTests.java (hunk 2 above)
Comment (asiajanicka) [reply src/test/java/org/example/parserTest/DOMTests.java]: I removed this info, and added some logging after creating schema and document objects and converting document to list of student groups.

File: src/test/java/org/example/parserTest/DOMTests.java (hunk 4 above)
Comment (asiajanicka) [reply src/test/java/org/example/parserTest/DOMTests.java:7]: I used it to create schema and validate xml (as I don't remember Andrew to show how to do it), later I learned that I could've used java.xml.validation instead.

**Pull request merged**